Automattic/wp-calypso
# Stats: Adds data to subscribers overview section

Related to #75928 

This PR includes improvements to the Subscribers Overview section. The changes made are outlined as follows:

- adds data from the mock API endpoint into the subscriber's overview cards
-  moved from making a single API request for all 90 days of stats and filtering out the required data to making individual queries for each required day (0, 30, 60, and 90 days ago). This approach, while increasing the number of requests, significantly reduces the amount of unnecessary data queried.
- migrated the section and react-query hook from JavaScript to TypeScript


## Testing Instructions

* Open the live branch for this PR.
* Navigate to `/stats/day/:siteSlug`.
* Add `?flags=stats/subscribers-section` to the URL to enable the subscriber stats page.
* Scroll down to locate the overview cards (immediately under the chart)
* Check that they load data correctly, still behave acceptably on resize etc. 
 
![image](https://github.com/Automattic/wp-calypso/assets/30754158/6cf80c05-2486-4772-88ed-f4f796154772)


## Pre-merge Checklist
